### PR TITLE
fix #20857 UncheckedArray[T] don't have default value and shouldn't be initialized

### LINKED ISSUE #20857: `UncheckedArray[T]` don't have default value and shouldn't be initialized
### What happened?

With this code snippet
```nim
type
  Obj = object
    s: string
    case b: bool
    of false: discard
    of true:
      v: int
      a: UncheckedArray[char]

var o: ref Obj
new o
o[] = Obj()
```

nim compiler end up with this error
```
Error: internal error: genAssignment: tyUncheckedArray
```

With this code snippet, it can prevent nim compiler from been terminated with the error above. but instead, nim compiler will generate C code to initialize the `UncheckedArray[T]` field
```nim
type
  Obj = object
    s: string
    case b: bool
    of false: discard
    of true:
      v: int
      a: UncheckedArray[char]

var o: ref Obj
new o
{.cast(uncheckedAssign).}:
  o.s = ""
  o.b = true
  o.v = 123
```

C compiler reports
```shell
/home/user/.cache/nim/x_d/@mx.nim.c: In function ‘eqdestroy___x_54’:
/home/user/.cache/nim/x_d/@mx.nim.c:262:34: error: invalid use of flexible array member
  262 |                 (*dest_2)._b_2.a = 0;
      |                                  ^
Error: execution of an external compiler program 'gcc -c  -w -fmax-errors=3 -pthread   -I/home/user/Projects/Nim/lib -I/home/user/Projects/Nim -o /home/user/.cache/nim/x_d/@mx.nim.c.o /home/user/.cache/nim/x_d/@mx.nim.c' failed with exit code: 1
```

With `refc`, compile and run without problem.

### Nim Version

1.2..devel, with arc/orc

### Current Standard Output Logs

```shell
Error: internal error: genAssignment: tyUncheckedArray
```

### Expected Standard Output Logs

Compile and run without errors

### Possible Solution

Don't try to initialize, copy or assign to `UncheckedArray[T]`.

### PR DESCRIPTION
**Pull request by bung87**
fix #20857

refs: https://github.com/nim-lang/Nim/issues/20788

### CODE REVIEW

File: compiler/ccgexprs.nim [hunk 1]
```diff
@@ -413,6 +413,8 @@ proc genAssignment(p: BProc, dest, src: TLoc, flags: TAssignmentFlags) =
   of tyPtr, tyPointer, tyChar, tyBool, tyEnum, tyCstring,
      tyInt..tyUInt64, tyRange, tyVar, tyLent, tyNil:
     linefmt(p, cpsStmts, "$1 = $2;$n", [rdLoc(dest), rdLoc(src)])
+  of tyUncheckedArray:
+    localError(p.config, dest.lode.info, "assignment of UncheckdArray is not supported with ARC/ORC")
```
Comment (tersec) [inline compiler/ccgexprs.nim]: `UncheckedArray`

File: tests/errmsgs/t20857.nim [hunk 2]
```diff
@@ -0,0 +1,18 @@
+
+discard """
+  matrix: "--gc:arc; --gc:orc"
+  errormsg: "assignment of UncheckdArray is not supported with ARC/ORC"
```
Comment (tersec) [inline tests/errmsgs/t20857.nim]: `UncheckedArrray`

File: compiler/ccgexprs.nim [hunk 3]
```diff
@@ -413,6 +413,8 @@ proc genAssignment(p: BProc, dest, src: TLoc, flags: TAssignmentFlags) =
   of tyPtr, tyPointer, tyChar, tyBool, tyEnum, tyCstring,
      tyInt..tyUInt64, tyRange, tyVar, tyLent, tyNil:
     linefmt(p, cpsStmts, "$1 = $2;$n", [rdLoc(dest), rdLoc(src)])
+  of tyUncheckedArray:
+    localError(p.config, dest.lode.info, "assignment of UncheckedArray is not supported with ARC/ORC")
```
Comment (Araq) [inline compiler/ccgexprs.nim:417]: It never was supported, this is not ARC/ORC specific!

Comment (bung87) [reply compiler/ccgexprs.nim:417]: it compile and runs on `refc`, and `refc` doesn't go in here.

Comment (Araq) [reply compiler/ccgexprs.nim:417]: it only happened to work with the other GCs. It was never supported as it never made sense.

Comment (bung87) [reply compiler/ccgexprs.nim:417]: so what's the error message would like

**Pull request closed without merging**